models/articles.model.js

```diff
@@ -14,6 +20,7 @@ exports.selectArticles = async () => {
                                            LEFT JOIN (SELECT article_id, COUNT(comment_id) as comment_count
                                                       FROM comments
                                                       GROUP BY article_id) c on c.article_id = a.article_id
+                                      ${whereClause}
                                   ORDER BY a.created_at desc;`);
 
   return results.rows.map((row) => ({...row, comment_count: +row.comment_count}));
```

Comment:
results.rows should already be an array including your comment_count, according to your sql call. Have a look at what is actually being logged here-> I think you might be adding an extra comment_count ??? But honestly I'm not sure 🤷   If it's not bringing the comment_count back, I'm unsure what your left join is doing

Reply:
The articles table does not include a comment count column, so the join is required in this case.